 <div style="display:inline-block; position:absolute; top:0; right:0; color: #230e3e; font:italic 1em serif;">$\LaTeX$<span style="display:inline-block; vertical-align:bottom">
    <a href="https://www.mathjax.org">
    <img style="height:1.5em" title="Powered by MathJax"
    src="https://www.mathjax.org/badge/mj_logo.png"
    border="0" alt="Powered by MathJax" >
</a></span></div>

<div style="display:inline-block; position:absolute; top:0; left:0; color: #230e3e; font:italic 1em serif;"><a href="https://www.julialang.org">
    <img style="height:2.25em; vertical-align:top" title="Powered by Julia"
    src="./figures/julia-logo-mask.png"
    border="0" alt="Powered by Julia" >
</a></div>
</a></div>$\require{cancel} \require{mhchem} \require{bbox} \require{color} \require{enclose} \require{AMSmath} \require{AMSsymbols}$

### Given
A mass of 5 kg of saturated liquid–vapor mixture of water is contained in a piston–cylinder device at 125 kPa. Initially, 2 kg of the water is in the liquid phase and the rest is in the vapor phase. Heat is now transferred to the water, and the piston, which is resting on a set of stops, starts moving when the pressure inside reaches 300 kPa. Heat transfer continues until the total volume increases by 20 percent. 

### Find
Determine  
  $\quad$(a) the initial and final temperatures,  
  $\quad$(b) the mass of liquid water when the piston first starts moving,  
  $\quad$(c) the work done during this process, and  
  $\quad$(d) show the process on a P-v diagram.


### Get Organized   

First, we break the problem down into states and the processes that connect them. We are given an initial state. Heat is added at constant volume until we reach the second state, which is right before the piston lifts. More heat is added at constant pressure until we reach the final state.

We also note that the only work interactions are boundary work.

![Cengelboles7ed4-126](./figures/Cengelboles7ed4-126.png)

Plotting the processes is also very helpful, and a habit you want to pick up.

![Cengelboles7ed4-126](./figures/Cengelboles7ed4-126plot.png)



Let's go through all the information we need to find in order to answer the questions.  

#### (a) the initial and final temperatures.  

$\quad$ $T_1 = T_{sat@125\,\rm kPa}$   
$\quad$ We get $T_3$ once we resolve State ⓷


#### (b) the mass of liquid water when the piston first starts moving,   
$\quad$ This is state ⓶, and $\displaystyle m_{liquid,2} = m_{total} - m_{vapor,2} = m_{total} - x_2 m_{total} = \left(1 - x_2 \right)m_{total}$

#### (c) the work done during this process   
$\quad$ $W_{total} = W_{1 \rightarrow 2} + W_{2 \rightarrow 3}$   

$\quad$ States ⓵ and ⓶ are connected by a constant volume process. With no volume change, there is no boundary work, so  $W_{1 \rightarrow 2} = 0$.

$\quad$ Thus, $$ W_{total} = W_{2 \rightarrow 3} = \int^3_2 P_2\, \text dV = P_2 \left( V_3 - V_2 \right) = P_2m \left( v_3 - v_2 \right) = P_2m \left( 1.2 v_1 - v_1 \right) = 0.2 P_2 m v_1 $$  


#### (d) show the process on a P-v diagram.
$\quad$ See above



### Attack 

 Today, we'll use Julia, with _Unitful.jl_, a package for calculation with physical units.

In [1]:
using Unitful       # Unitful units package
const kJ = u"kJ"    # make shortcuts to Unitful units constructor
const kg = u"kg"
const m = u"m"
const kPa = u"kPa"
const °C = u"°C";

#### (a) the initial and final temperatures.  
$\quad$ $\color{red} T_1 = T_{sat@125\,\rm kPa} =   105.52\, ℃ $ 

$\quad$ We'll come back to $T_3.$

#### (b) the mass of liquid water when the piston first starts moving     
$\quad$ $m_{liquid,2} = \left(1 - x_2 \right)m_{total}$, so we need $x_2: \ce{H2O} \left\{ \begin{align*} \ce{P_2 \ &= \ 300 kPa} \\
    \ce{v_2 \ &= \ v_1 }
    \end{align*} 
    \right.$

$\quad$ At State ⓵, $\ce{H2O} \left\{ \begin{align*} \ce{P_1 \ &= \ 125 kPa} \\
    x_1 \ &= \ 0.6 
    \end{align*} 
    \right.$   
    

In [2]:
vf_1 = (1.0432e-3 + 1.0528e-3)m^3/kg   # interpolate with target halfway between table breakpoints
vg_1 = (1.694 + 1.159)m^3/kg
x_1 = 3kg / 5kg

v_1 = vf_1 + x_1 * (vg_1 - vf_1)

1.7126384 m^3 kg^-1

$\quad$ Back to finding $x_2: \ce{H2O} \left\{ \begin{align*} \ce{P_2 \ &= \ 300 kPa} \\
    \ce{v_2 \ &= \ v_1} = 1.713\ m^3/kg
    \end{align*} 
    \right.$  

$\quad$ $v_2 = 0.6058\, m^3/kg. \quad  v_{g@300kPa} = 0.6058\, m^3/kg. \quad v_2 > v_{g@300kPa}.$   

$\quad$ Therefore, we have superheated vapor, and $ \color{red} m_{liquid,2} = 0.$

#### (c) the work done during this process   
$\quad$  $W_{total} = 0.2 P_2 m v_1$   

In [3]:
# Calculations
mass = 5kg
P_2 = 300kPa
W = 0.2 * P_2 * mass * v_1
W |> kJ

513.79152 kJ

$\quad$ $\color{red} W_{total} = 513.79\, kJ$   

#### back to (a) and $T_3$.  

In [4]:
v_3 = 1.2 * v_1

2.05516608 m^3 kg^-1

$\quad$ Finding $T_3: \ce{H2O} \left\{ \begin{align*} \ce{P_3 \ &= \ 300 kPa} \\
    v_3 \ &= 2.055\ \rm m^3/kg
    \end{align*} 
    \right.$  

$\quad$ Unfortuately, the steam tables in your textbook do not include state ③, so I'll use different property tables to solve.  

| T [℃]   | v [$m^3/kg$]  |
| --------|:-------------:|
| 1000    | 1.95824       |  
| 1100    | 2.11226       | 


In [5]:
# not in the Moran/Shapiro textbook tables
# interpolate
T_1000 = 1000°C
T_1100 = 1100°C
v_1000 = 1.95824m^3/kg
v_1100 = 2.11226m^3/kg

T_3 = T_1000 + (T_1100 - T_1000)*(v_3 - v_1000)/(v_1100 - v_1000)

1062.9308401506296 °C

$\quad$ $\color{red} T_3 = 1062.93\, \rm °C$ 

<footer style="background-color: #461d7c; color:rgba(253,208,35,0.35); line-height:2em; padding-left:1em; padding-right:0.5em; bottom:0; float:bottom; text-align:right;"><i>ψᵟ</i></footer>